neuraxle/base.py

```diff
@@ -472,35 +725,131 @@ def set_savers(self, savers: List[BaseSaver]) -> 'BaseStep':
+        """
+        Get step hyperparameters as :class:`HyperparameterSamples`.
+
+        :return: step hyperparameters
+        :rtype: HyperparameterSamples
+
+        .. seealso::
+            * :class:`HyperparameterSamples`
+        """
         return self.hyperparams
 
     def set_params(self, **params) -> 'BaseStep':
+        """
+        Set step hyperparameters with a dictionary.
+
+        Example :
+        ```
+        s.set_params(learning_rate=0.1)
+        hyperparams = s.get_params()
+        assert hyperparams == {"learning_rate": 0.1}
+        ```
+
+        :param **params: arbitrary number of arguments for hyperparameters
+        :rtype: BaseStep
+
+        .. seealso::
+            * :class:`HyperparameterSamples`
+        """
         return self.set_hyperparams(HyperparameterSamples(params))
 
     def get_params(self) -> dict:
+        """
+        Get step hyperparameters as a flat primitive dict.
+
+        Example :
+        ```
+        s.set_params(learning_rate=0.1)
+        hyperparams = s.get_params()
+        assert hyperparams == {"learning_rate": 0.1}
+        ```
+
+        :return: hyperparameters
+        :rtype: dict
+
+        .. seealso::
+            * :class:`HyperparameterSamples`
+        """
         return self.get_hyperparams().to_flat_as_ordered_dict_primitive()
 
     def set_hyperparams_space(self, hyperparams_space: HyperparameterSpace) -> 'BaseStep':
+        """
+        Set step hyperparameters space.
+
+        Example :
+        ```
+        step.set_hyperparams_space(HyperparameterSpace({
+            'hp': RandInt(0, 10)
+        }))
+        ```
+
+        :param hyperparams_space: hyperparameters space
+        :type hyperparams_space: HyperparameterSpace
+        :return: self
+        :rtype: BaseStep
+
+        .. seealso::
+            * :class:`HyperparameterSpace`
+            * :class:`HyperparameterDistribution`
+        """
         self.is_invalidated = True
         self.hyperparams_space = HyperparameterSpace(hyperparams_space).to_flat()
         return self
 
     def get_hyperparams_space(self) -> HyperparameterSpace:
+        """
+        Get step hyperparameters space.
+
+        Example :
+        ```
+        step.get_hyperparams_space()
+        ```
+
+        :return: step hyperparams space
+        :rtype: HyperparameterSpace
+
+        .. seealso::
+            * :class:`HyperparameterSpace`
+            * :class:`HyperparameterDistribution`
+        """
         return self.hyperparams_space
 
     def handle_fit(self, data_container: DataContainer, context: ExecutionContext) -> ('BaseStep', DataContainer):
         """
-        Perform any needed side effects on the step or the data container before fitting the step.
+        Override this to add side effects or change the execution flow before (or after) calling * :func:`~neuraxle.base.BaseStep.fit`.
+        The default behavior is to rehash current ids with the step hyperparameters.
```

Comment:
The handle methods will need to be more documented in the future to explain that using handle methods to hash things is in the only purpose of checkpointing data and managing execution flow from data containers.

Reply:
See also :class DataContainer and Pipeline